diffkemp/simpll/simpll.py

```diff
@@ -21,52 +21,100 @@ def add_suffix(file, suffix):
     return "{}-{}{}".format(name, suffix, ext)
 
 
-def run_simpll(first, second, fun_first, fun_second, var, suffix=None,
-               cache_dir=None, pattern_config=None, control_flow_only=False,
-               output_llvm_ir=False, print_asm_diffs=False, verbosity=0,
-               use_ffi=False, module_cache=None, modules_to_cache=None):
+def run_simpll(first, second, fun_first, fun_second, var, config, suffix=None,
+               cache_dir=None, module_cache=None, modules_to_cache=None):
     """
     Simplify modules to ease their semantic difference. Uses the SimpLL tool.
     :return A tuple containing the two LLVM IR files generated by SimpLL
             followed by the result of the comparison in the form of a graph and
             a list of missing function definitions.
     """
     stderr = None
-    if verbosity == 0:
+    if config.verbosity == 0:
         stderr = open(os.devnull, "w")
 
     first_out_name = add_suffix(first, suffix) if suffix else first
     second_out_name = add_suffix(second, suffix) if suffix else second
 
-    if use_ffi:
+    if config.use_ffi:
+        builtin_pattern_args = ffi.new("struct builtin_pattern_args *")
```

Comment:
The below is quite a chunk of code. Separate into function?

Reply:
I resolved this using the method `as_ffi_struct` of the `BuiltinPatterns` class in the updated version of the PR.